src/Common/src/System/Security/Cryptography/KeyBlobHelpers.cs

```diff
@@ -10,6 +10,45 @@ namespace System.Security.Cryptography
 {
     internal static class KeyBlobHelpers
     {
+        internal static byte[] ToUnsignedIntegerBytes(this ReadOnlyMemory<byte> memory, int length)
+        {
+            if (memory.Length == length)
+            {
+                return memory.ToArray();
+            }
+
+            ReadOnlySpan<byte> span = memory.Span;
+
+            if (memory.Length == length + 1)
+            {
+                if (span[0] == 0)
+                {
+                    return span.Slice(1).ToArray();
+                }
+            }
+
+            if (span.Length > length)
+            {
+                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+            }
+
+            byte[] target = new byte[length];
+            span.CopyTo(target.AsSpan(length - span.Length));
+            return target;
+        }
+
+        internal static byte[] ToUnsignedIntegerBytes(this ReadOnlyMemory<byte> memory)
```

Comment:
would it make sense to just make length optional in the other overload?

Reply:
It could be defaulted to -1, I suppose, but that is really a different method with a different purpose.  This is "remove an optional padding byte, retain your length as self-contained and/or semantic".  The other is "this answer must be precisely this big, and here are the transforms we have to do to it"